nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/json/AbstractJsonRowRecordReader.java

```diff
@@ -197,7 +193,8 @@ private JsonNode getNextJsonNode() throws JsonParseException, IOException, Malfo
                     return jsonParser.readValueAsTree();
                 case END_ARRAY:
                 case START_ARRAY:
-                    return null;
+                    return getNextJsonNode();
```

Comment:
Would recommend that we just use 'continue' here as we do for END_OBJECT, rather than recursively calling ourselves. Is consistent and avoids unnecessarily deepening the stack, but will result in the same logic being evaluated